src/CashuWallet.ts

```diff
@@ -118,14 +121,11 @@ class CashuWallet {
 					mintKeys.set(token.mint, keys);
 				}
 			} catch (error) {
-				// I'm not sure if this is the best way to handle errors
-				if (proofs.length) {
-					return proofs;
-				}
-				throw error;
+				console.error(error);
+				tokensWithErrors.push(token);
 			}
 		}
-		return proofs;
+		return { proofs, tokensWithErrors };
```

Comment:
should we return `tokensWithErrors` as a `Token` object? so that the client doesn't have to construct it himself. We can then only return it if `tokensWithErrors.tokens.length is > 0`  So we can set the `memo` as well in the returned object, and if not, return `undefined`/`null`.